src/features/courses/course-controller.ts

```diff
@@ -1217,10 +1217,37 @@ class CourseController {
             // when no workbook is sent, the source of truth depends on whether question belongs to an assessment
             const thisQuestionIsFromAnAssessment = await this.isQuestionAnAssessment(options.questionId);
             if (thisQuestionIsFromAnAssessment) {
-                gradeInstance = await courseRepository.getCurrentInstanceForQuestion({
-                    questionId: options.questionId, 
-                    userId: options.userId
-                }); 
+                if (_.isNil(options.studentTopicAssessmentInfoId)) {
+                    gradeInstance = await courseRepository.getCurrentInstanceForQuestion({
+                        questionId: options.questionId,
+                        userId: options.userId
+                    }); 
+                } else {
+                    // student grade instances do not have a direct reference to userId
+                    // we have to pass through the studentGrade to get there
+                    const questionGrade = await courseQuestion.getGrades({
+                        where: {
+                            userId: options.userId
```

Comment:
Should always have active true, though there is no way to set them as inactive right now